taichi/rhi/vulkan/vulkan_device.cpp

```diff
@@ -2405,28 +2413,14 @@ void VulkanSurface::create_swap_chain() {
   image_format_ = buffer_format_vk_to_ti(surface_format.format);
 
   for (VkImage img : swapchain_images) {
-    VkImageViewCreateInfo view_info{};
-    view_info.sType = VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO;
-    view_info.pNext = nullptr;
-    view_info.image = img;
-    view_info.viewType = VK_IMAGE_VIEW_TYPE_2D;
-    view_info.format = surface_format.format;
-    view_info.components.r = VK_COMPONENT_SWIZZLE_IDENTITY;
-    view_info.components.g = VK_COMPONENT_SWIZZLE_IDENTITY;
-    view_info.components.b = VK_COMPONENT_SWIZZLE_IDENTITY;
-    view_info.components.a = VK_COMPONENT_SWIZZLE_IDENTITY;
-    view_info.subresourceRange.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
-    view_info.subresourceRange.baseMipLevel = 0;
-    view_info.subresourceRange.levelCount = 1;
-    view_info.subresourceRange.baseArrayLayer = 0;
-    view_info.subresourceRange.layerCount = 1;
-
-    vkapi::IVkImage image = vkapi::create_image(device_->vk_device(), img);
+    vkapi::IVkImage image = vkapi::create_image(
+        device_->vk_device(), img, surface_format.format, VK_IMAGE_TYPE_2D,
+        VkExtent3D{uint32_t(width), uint32_t(height), 1}, 1u, 1u, usage);
     vkapi::IVkImageView view =
-        vkapi::create_image_view(device_->vk_device(), image, &view_info);
+        vkapi::create_image_view(device_->vk_device(), image);
```

Comment:
`vkapi::` is our light-weight Vulkan C++ reference counted header wrapper, so we should make it adhereing as close to the Vulkan API as possible, this includes using the original Vulkan create info struct if it contains more stuff than we currently use. (i.e. Maybe we want to change swizzle flags, maybe we want a view that is not starting from layer 0)

Reply:
I'd avoid making "smart" wrappers like this one here, also makes the code harder to understand